libs/python/detect.go

```diff
@@ -32,3 +46,29 @@ func DetectExecutable(ctx context.Context) (string, error) {
 	}
 	return interpreter.Path, nil
 }
+
+// DetectVEnvExecutable returns the path to the python3 executable inside venvPath,
+// that is not necessarily activated.
+//
+// If virtualenv is not created, or executable doesn't exist, the error is returned.
+func DetectVEnvExecutable(venvPath string) (string, error) {
+	interpreterPath := filepath.Join(venvPath, "bin", "python3")
+	if runtime.GOOS == "windows" {
+		interpreterPath = filepath.Join(venvPath, "Scripts", "python3.exe")
+	}
+
+	if _, err := os.Stat(interpreterPath); err != nil {
+		if os.IsNotExist(err) {
+			return "", fmt.Errorf("can't find %q, check if virtualenv is created", interpreterPath)
+		} else {
+			return "", fmt.Errorf("can't find %q: %w", interpreterPath, err)
+		}
+	}
+
+	_, err := os.Stat(filepath.Join(venvPath, "pyvenv.cfg"))
+	if errors.Is(err, fs.ErrNotExist) {
+		return "", fmt.Errorf("expected %q to be virtualenv, but pyenv.cfg is missing", venvPath)
+	}
```

Comment:
What happens on other errors?

Reply:
Not sure what else can go wrong. What do you mean?

Reply:
Any other error, like `fs.ErrPermission`.

Like the conditional above, it's good practice to never swallow errors.

Reply:
👍 Oh, I see. This is a mistake, it wasn't intended. I will fix it